.github/workflows/workflow.yml

```diff
@@ -124,16 +124,17 @@ jobs:
           toolchain: stable
           override: true
 
-      # Install Python at a fixed version
-      - uses: actions/setup-python@v1
-        with:
-          python-version: "3.7.7"
-
       # Install dotnet at a fixed version
       - uses: actions/setup-dotnet@v1
         with:
           dotnet-version: "2.2.402"
 
+      # Ensure we have a recent version of python installed for Mercurial
+      - uses: actions/setup-python@v1
+        if: matrix.os == 'macOS-latest'
+        with:
+          python-version: "3.8"
+
```

Comment:
I think we don't even need that since [mercurial requires python 3.5](https://www.mercurial-scm.org/wiki/SupportedPythonVersions) and [python 3.7 is preinstalled](https://github.com/actions/virtual-environments/blob/master/images/macos/macos-10.15-Readme.md#language-and-runtime).